packages/terra-disclosure-manager/src/DisclosureManager.jsx

```diff
@@ -240,17 +240,8 @@ class DisclosureManager extends React.Component {
      * Allows a component to register a function with the DisclosureManager that will be called before the component is dismissed for any reason.
      */
     delegate.registerDismissCheck = (checkFunc) => {
-      const { disclosureManager } = this.props;
-
       this.dismissChecks[componentData.key] = checkFunc;
 
-      if (disclosureManager && disclosureManager.registerDismissCheck) {
```

Comment:
Are people still allowed to register a custom dismiss check?

Reply:
They can. The dismiss check just never leaves the scope of the DisclosureManager it is directly registered to. 

The removed logic would take the provided function and register it to an ancestor DisclosureManager, if one exists. This would have happened if, for example, a SlidePanelManager was rendered within a ModalManager modal. Not exactly an expected or UX desired configuration, but one that exposed these issues.

Reply:
I will say that the NavigationPrompt is the recommended and future-facing solution to dismissal prevention. registerDismissCheck is on the chopping block whenever we decide to revisit our disclosure patterns.